### PR TITLE
Backport timepicker to v2

### PR DESCRIPTION
**Pull request by GilbertCherrie**
This PR allows for the back port of the time picker component changes to v2.

### CODE REVIEW

**Comment by GilbertCherrie** [comment]
@rvsia these are all the file changes related to the time picker changes that need back porting. However I am not sure if anything is missing from these changes that needs to be included in the back port.

**Comment by rvsia** [comment]
@GilbertCherrie the v2 has a flat file structure. All the component files should be in `files` folder. You do not need to include index files.

**Comment by GilbertCherrie** [comment]
@rvsia I changed it to a flat structure. Please let me know if anything else needs to be fixed.

File: packages/carbon-component-mapper/src/files/time-pircker-date.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { UseFieldApiComponentConfig } from "@data-driven-forms/react-form-renderer";
```
Comment (rvsia) [inline packages/carbon-component-mapper/src/files/time-pircker-date.d.ts:1]: Not sure if it is an issue in the original file, but there is a typo in the filename, could you please fix it? I will check v3 later

File: packages/carbon-component-mapper/src/tests/time-picker.test.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
 import { mount } from 'enzyme';
 
-import FormRenderer, { componentTypes } from '@data-driven-forms/react-form-renderer';
+import { FormRenderer, componentTypes } from '@data-driven-forms/react-form-renderer';
 
-import FormTemplate from '../files/form-template';
-import componentMapper from '../files/component-mapper';
+import FormTemplate from '../form-template';
+import componentMapper from '../component-mapper';
```
Comment (rvsia) [inline packages/carbon-component-mapper/src/tests/time-picker.test.js:7]: These imports should not be changed

**Comment by GilbertCherrie** [comment]
Made the necessary fixes and have repushed.

File: packages/carbon-component-mapper/src/tests/time-picker.test.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { mount } from 'enzyme';
 
-import FormRenderer, { componentTypes } from '@data-driven-forms/react-form-renderer';
+import { FormRenderer, componentTypes } from '@data-driven-forms/react-form-renderer';
```
Comment (rvsia) [inline packages/carbon-component-mapper/src/tests/time-picker.test.js:4]: Oh, this will probably also cause issues in tests.

Comment (GilbertCherrie) [reply packages/carbon-component-mapper/src/tests/time-picker.test.js:4]: Is the original line correct?

Comment (rvsia) [reply packages/carbon-component-mapper/src/tests/time-picker.test.js]: Yeah, in v2 FormRenderer is exported as the default

**Comment by rvsia** [comment]
@GilbertCherrie Tests are faling

```jsx
Summary of all failing tests
 FAIL  packages/carbon-component-mapper/src/tests/time-picker.test.js
  ● TimePicker › handle invalid date

    expect(received).toEqual(expected) // deep equality

    Expected: "24:00"
    Received: "00:00"

       99 |     wrapper.update();
      100 | 
    > 101 |     expect(wrapper.find('input').props().value).toEqual('24:00');
          |                                                 ^
      102 |     expect(onSubmit.mock.calls[0][0]['time-picker'].getHours()).toEqual(0);
      103 |     expect(onSubmit.mock.calls[0][0]['time-picker'].getMinutes()).toEqual(0);
      104 |   });

      at _callee10$ (packages/carbon-component-mapper/src/tests/time-picker.test.js:101:49)
      at tryCatch (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:45:40)
      at Generator.invoke [as _invoke] (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:271:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:97:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)

  ● TimePicker › handles initial value

    TypeError: Cannot read property 'getHours' of undefined

      244 |     wrapper.update();
      245 | 
    > 246 |     expect(onSubmit.mock.calls[0][0]['time-picker'].getHours()).toEqual(15);
          |                                                     ^
      247 |     expect(onSubmit.mock.calls[0][0]['time-picker'].getMinutes()).toEqual(0);
      248 |   });
      249 | });

      at _callee28$ (packages/carbon-component-mapper/src/tests/time-picker.test.js:246:53)
      at tryCatch (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:45:40)
      at Generator.invoke [as _invoke] (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:271:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:97:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
```

**Comment by GilbertCherrie** [comment]
> @GilbertCherrie Tests are faling
> 
> ```js
> Summary of all failing tests
>  FAIL  packages/carbon-component-mapper/src/tests/time-picker.test.js
>   ● TimePicker › handle invalid date
> 
>     expect(received).toEqual(expected) // deep equality
> 
>     Expected: "24:00"
>     Received: "00:00"
> 
>        99 |     wrapper.update();
>       100 | 
>     > 101 |     expect(wrapper.find('input').props().value).toEqual('24:00');
>           |                                                 ^
>       102 |     expect(onSubmit.mock.calls[0][0]['time-picker'].getHours()).toEqual(0);
>       103 |     expect(onSubmit.mock.calls[0][0]['time-picker'].getMinutes()).toEqual(0);
>       104 |   });
> 
>       at _callee10$ (packages/carbon-component-mapper/src/tests/time-picker.test.js:101:49)
>       at tryCatch (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:45:40)
>       at Generator.invoke [as _invoke] (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:271:22)
>       at Generator.prototype.(anonymous function) [as next] (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:97:21)
>       at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
>       at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
> 
>   ● TimePicker › handles initial value
> 
>     TypeError: Cannot read property 'getHours' of undefined
> 
>       244 |     wrapper.update();
>       245 | 
>     > 246 |     expect(onSubmit.mock.calls[0][0]['time-picker'].getHours()).toEqual(15);
>           |                                                     ^
>       247 |     expect(onSubmit.mock.calls[0][0]['time-picker'].getMinutes()).toEqual(0);
>       248 |   });
>       249 | });
> 
>       at _callee28$ (packages/carbon-component-mapper/src/tests/time-picker.test.js:246:53)
>       at tryCatch (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:45:40)
>       at Generator.invoke [as _invoke] (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:271:22)
>       at Generator.prototype.(anonymous function) [as next] (node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:97:21)
>       at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
>       at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
> ```

@rvsia I fixed the first case by changing 24 to 00:00 but not sure what is wrong with the second case. Do you have any idea what's wrong here or how I can fix it?

**Comment by GilbertCherrie** [comment]
This second failing test looks the same as the test for handle change, except for some reason it is not passing. Not sure what the issue is with this test.

**Comment by rvsia** [comment]
@GilbertCherrie I don't know. Seems that the value is not stored in submit. It would be better to debug it live - I will take a look.

**Comment by rvsia** [comment]
@GilbertCherrie  I found it.

In v2 there is a bug with date as an initial value, the bug is fixed in v3. 

Can you please add this PR here https://github.com/data-driven-forms/react-forms/pull/1110

**Comment by GilbertCherrie** [comment]
@rvsia I made the fixes and have pushed. Please let me know if anything else needs changing.

**Comment by rvsia** [comment]
@GilbertCherrie tests passed but there are many lint errors, run `yarn lint` to see them and `yarn lint --fix` to fix them.

**Comment by GilbertCherrie** [comment]
yarn lint does not seem to work, it runs but nothing seems to happen. Might have to fix them manually.

**Comment by rvsia** [comment]
@GilbertCherrie works fine for me 🤔  have you ran `yarn` to install dependencies? You have to be on node10 (`sudo n 10` to switch)

**Comment by GilbertCherrie** [comment]
@rvsia Pushed lint changes.

**Review by rvsia — APPROVED** [review]

**Comment by GilbertCherrie** [comment]
@rvsia Thank you for the help and quick merge.

**Comment by rvsia** [comment]
Released: 

`@data-driven-forms/carbon-component-mapper@2.24.11`
`@data-driven-forms/react-form-renderer@2.24.2`

**Pull request merged**